avocado/core/loader.py

```diff
@@ -694,8 +693,7 @@ def ignore_broken(klass, uid, params=None):
                                                 subtests_filter)
             else:
                 if os.access(test_path, os.X_OK):
-                    return self._make_test(test.SimpleTest,
-                                           pipes.quote(test_path))
+                    return self._make_test(test.SimpleTest, test_path)
```

Comment:
Yes, it's not optimal, but very handy. Right now in my CI I use `external-runner`, so I can cope with that, but it'd be impossible to combine different kind of tests together. What am I talking about is:

```
avocado run "./run 1" "./run 2" ... "./run 999"
```

can be turned into

```
avocado run --external-runner ./run 1 2 ... 999
```

but if I wanted to run various kinds of tests in one job

```
avocado run boot "./run 1" "./run 2" ... "./run 999" other_suite.py
```

I would have to use SimpleTest with args.

Right now I use different jobs to run functional, qemu-iotests and qemu-kvm-unittests, but who knows, I'd like to have the possibility.

On the other hand yes, it's not nice, it actually requires running the discovery twice for each reference so maybe we could remove it. But first we need to be pretty sure nobody uses it (or can replace it). Alternative solution would be to implement the test-type prefix as we discussed several times and allow SimpleTestWithArgs only when it's specified:

```
avocado run VT://boot "SimpleArgs://foo --bar --baz" passtest.py
```

where without the prefix it uses autodetect (as it works now), with the prefix it only involves the specified loader. The `SimpleArgs` loader would be disabled for autodetection (similar to external-runner).